### DISCUSSION TITLE
Thousands of folders starting with "wsl" added to the root of the Ubuntu installation

### DISCUSSION THREAD

**Discussion by k-risc** [microsoft/WSL#8911] [Q&A]
So far it only happened two times, with a longer period in between, but as the last time this happened was just yesterday, I wonder what the reasons might be.
I recognized that under the root of the Ubuntu installation, I suddenly have thousands of folders, all staring with "wsl" and then seemingly some has value (wslAaabIJ, wslAabLaG, ..., wslPppJCm). They all have been added yesterday morning within nine minutes.

I remember that I recognized this some weeks ago once, and back then I removed the folders from within Ubuntu (was not able from outside, from Windows), using sudo rights. It did not have any impact. As far as I remember, I spot checked some of them before, and they were all empty.

I cannot remember whether anything out of the ordinary happened the last time, but yesterday monbing when I wanted to start working it said that there was some fatal exception with WSL and it did not work. consequently, when trying to access the Ubuntu folders via Windows explorer, when trying to open an Ubuntu shell, but also when trying to open PhpStorm (with a project residing in Ubuntu WSL), this all did not work. After a restart of my computer, everything worked again. And today I recognize these thousands of folders.

**Comment by cerebrate** [comment]
I've never had thousands, but I have had these empty folders showing up randomly for quite some time. Afraid I never did figure out what was going on to cause them, except for a loose hypothesis that it had _something_ to do with using cross-distro mounting, and I never did confirm that. There's also #7152, but that never got definitive closure, either, and at least some of mine were definitely showing up post the mentioned fix.

(So as not to be totally unhelpful, I'll add that my cheap workaround is to create the following file as _/etc/tmpfiles.d/anti-wsl.conf_:

```
# Remove the empty /wsl* directories generated mysteriously.
r! /wsl*
```

and it clears them all up nicely when you restart WSL. Requires systemd support, whether official or hacked.)

**Reply by jbrazda** [reply]
Thanks! I had about a dozen of these, and I was not sure if you can just delete those. This solution works nicely.
